substratest/client.py

```diff
@@ -209,10 +209,7 @@ def wait(self, asset, timeout=cfg.FUTURE_TIMEOUT, raises=True):
             assert False, 'Future not supported'
         getter = getattr(self, m)
 
-        if asset.__class__.__name__ == "ComputePlan":
-            key = asset.compute_plan_id
-        else:
-            key = asset.key
+        key = asset.key
```

Comment:
Can we just remove this variable and use asset.key everywhere?

Reply:
The only place where we use this variable is 

```
asset = getter(key)
```

I think changing it to 

```
asset = getter(asset.key)
```

might potentially bring new categories of bugs: if `getter` doesn't return the correct result, then the subsequent `getter` call will use the wrong argument. For this reason I think it's better to create a dedicated variable `key` to ensure `key` is constant throughout the execution of this function.

What do you think?

Reply:
Makes sense, thank you for the explanation!